app/utils/url/path.ts

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+export const ID_PATH_PATTERN = '[a-z0-9]{26}';
+export const PLUGIN_ID_PATH_PATTERN = '^[a-zA-Z0-9-_.]+$';
+
+// This should cover:
+// - Team name (lowercase english characters, numbers or -)
+// - Two ids separated by __ (userID__userID)
+export const TEAM_NAME_PATH_PATTERN = '[a-z0-9\\-_]+';
+
+// This should cover:
+// - Channel name
+// - Channel ID
+// - Group Channel Name (40 length UID)
+// - DM Name (userID__userID)
+// - Username prefixed by a @
+// - User ID
+// - Email
+export const IDENTIFIER_PATH_PATTERN = '[@a-zA-Z\\-_0-9][@a-zA-Z\\-_0-9.]*';
```

Comment:
Usernames can be non latin as well

Reply:
Same here, only English characters are allowed in usernames: https://github.com/mattermost/mattermost/blob/51715d5ed695efea3aa3ff86a07951ca0bf6177a/server/public/model/user.go#L917-L928

Email addresses could contain things not matched by this, but the email address matching supposedly isn't actually used for anything at least in the webapp, so that's ok.